sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/types/DataTypeUtils.scala

```diff
@@ -47,6 +47,14 @@ object DataTypeUtils {
     DataType.equalsIgnoreCaseAndNullability(from, to)
   }
 
+  /**
+   * Compares two types, ignoring nullability of ArrayType, MapType, StructType, ignoring case
+   * sensitivity of field names in StructType as well as differences in collation for String types.
+   */
+  def equalsIgnoreCaseNullabilityAndStringTypes(from: DataType, to: DataType): Boolean = {
```

Comment:
why do we need this extra level of indirection?

Reply:
both `equalsIgnoreNullability` and `equalsIgnoreCaseAndNullability` do the same thing so I just wanted to be consistent